gqcp/include/QCModel/CC/CCD.hpp

```diff
@@ -0,0 +1,449 @@
+                    for (const auto& e : orbital_space.indices(OccupationType::k_virtual)) {
+                        for (const auto& f : orbital_space.indices(OccupationType::k_virtual)) {
+                            value += 0.5 * tau2_tilde(i, n, e, f) * V_A(m, n, e, f);
+                        }
+                    }
+                }
+
+                F2(m, i) = value;
+            }
+        }
+
+        return F2;
+    }
+
+
+    /**
+     *  @param f                    the (inactive) Fock matrix
+     *  @param V_A                  the antisymmetrized two-electron integrals (in physicist's notation)
+     *  @param t2                   the T2-amplitudes
+     * 
+     *  @return the F3-intermediate
+     * 
+     *  @note This is one of the intermediate quantities in the factorization of CCD. In particular, F3 represents equation (5) in Stanton1991.
+     */
+    static ImplicitMatrixSlice<Scalar> calculateF3(const QCMatrix<Scalar>& f, const QCRankFourTensor<Scalar>& V_A, const T2Amplitudes<Scalar>& t2) {
+
+        const auto& orbital_space = t2.orbitalSpace();
+
+        // Implement the formula for F3 in equation (5) in Stanton1991.
+        auto F3 = orbital_space.template initializeRepresentableObjectFor<Scalar>(OccupationType::k_occupied, OccupationType::k_virtual);  // zero-initialize an occupied-virtual object
+        for (const auto& m : orbital_space.indices(OccupationType::k_occupied)) {
+            for (const auto& e : orbital_space.indices(OccupationType::k_virtual)) {
+                // Calculate the contribution from the first term.
+                F3(m, e) = f(m, e);
+            }
+        }
+
+        return F3;
+    }
+
+
+    /**
+     *  @param t2                   the T2-amplitudes
+     * 
+     *  @return the tau2-intermediate
+     * 
+     *  @note This is one of the intermediate quantities in the factorization of CCD. In particular, tau2 represents equation (10) in Stanton1991.
+     */
+    static ImplicitRankFourTensorSlice<Scalar> calculateTau2(const T2Amplitudes<Scalar>& t2) {
+
+        const auto& orbital_space = t2.orbitalSpace(); 
+
+        // Implement the formula for tau2 (equation 10).
+        auto tau2 = t2.asImplicitRankFourTensorSlice();  // the contribution from the first term
+
+        return tau2;
+    }
+
+
+    /**
+     *  @param t2                   the T2-amplitudes
+     * 
+     *  @return the tau2_tilde-intermediate
+     * 
+     *  @note This is one of the intermediate quantities in the factorization of CCD. In particular, tau2_tilde represents equation (9) in Stanton1991.
+     */
+    static ImplicitRankFourTensorSlice<Scalar> calculateTau2Tilde(const T2Amplitudes<Scalar>& t2) {
+
+        const auto& orbital_space = t2.orbitalSpace(); 
+
+        // Implement the formula for tau2_tilde (equation 9).
+        auto tau2_tilde = t2.asImplicitRankFourTensorSlice();  // the contribution from the first term
+
+        return tau2_tilde;
+    }
```

Comment:
I see this is the easiest way to copy-paste the current CCSD implementation and change it to CCD, but in CCD `tau2` and `tau2_tilde` *are* just `t2` I think it's better to delete these methods altogether.

Reply:
Yes, I wanted to change this, but wanted to see if everything else was working fine. Must've forgot to change this.